vars/unitTestPost.groovy

```diff
@@ -26,90 +26,111 @@
    * config['valgrind_stash']      Name to stash valgrind artifacts
    *                               Required if more than one stage is
    *                               creating valgrind reports.
-   *
    */
 
-def call(Map config = [:]) {
-
-  String always_script = config.get('always_script',
-                                    'ci/unit/test_post_always.sh')
-  sh label: 'Job Cleanup',
-     script: always_script
-
-  Map stage_info = parseStageInfo(config)
+// groovylint-disable DuplicateNumberLiteral, DuplicateStringLiteral,
+// groovylint-disable VariableName
 
-  if (config['testResults'] != 'None' ) {
-    double health_scale = 1.0
-    if (config['ignore_failure']) {
-      health_scale = 0.0
-    }
+// groovylint-disable-next-line MethodSize
+void call(Map config = [:]) {
+    String always_script = config.get('always_script',
+                                      'ci/unit/test_post_always.sh')
+    sh label: 'Job Cleanup',
+       script: always_script
 
-    def cb_result = currentBuild.result
-    junit testResults: config.get('testResults', 'test_results/*.xml'),
-          healthScaleFactor: health_scale
+    Map stage_info = parseStageInfo(config)
 
-    if (cb_result != currentBuild.result) {
-      println "The junit plugin changed result to ${currentBuild.result}."
+    if (config['testResults'] != 'None' ) {
+        // groovylint-disable-next-line NoDouble
+        double health_scale = 1.0
+        if (config['ignore_failure']) {
+            health_scale = 0.0
+        }
+        String cb_result = currentBuild.result
+        junit testResults: config.get('testResults', 'test_results/*.xml'),
+              healthScaleFactor: health_scale
+        if (cb_result != currentBuild.result) {
+            println "The junit plugin changed result to ${currentBuild.result}."
+        }
     }
-  }
-
-  if(stage_info['with_valgrind']) {
-    String target_dir = "unit_test_memcheck_logs"
-    String src_files = "unit-test-*.memcheck.xml"
-    fileOperations([fileCopyOperation(excludes: '',
+    if (stage_info['with_valgrind']) {
+        String target_dir = 'unit_test_memcheck_logs'
+        String src_files = 'unit-test-*.memcheck.xml'
+        int vgfail = 0
+        int vgerr = 0
+        String rcs = sh label: 'Check for Valgrind errors',
+                 script: "grep -E '<error( |>)' ${src_files} || true",
+                 returnStdout: true
+        if (rcs) {
+            vgfail = 1
+        }
+        String suite = sanitizedStageName()
+        junitSimpleReport suite: suite,
+                          file: suite + '_valgrind_results.xml',
+                          fails: vgfail,
+                          errors: vgerr,
+                          name: 'Valgrind_Memcheck',
+                          class: 'Valgrind',
+                          message: 'Valgrind detected',
```

Comment:
```suggestion
                          message: 'Valgrind error detected',
```

Reply:
Technically they're memcheck warnings.

Reply:
Fair enough.  Whatever the nomenclature is, but _Valgrind detected_ just reads incomplete.